### PR TITLE
Adds docs for custom targeting on mobile

### PR DESCRIPTION
**Pull request by tarikeshaq**
fixes https://mozilla-hub.atlassian.net/browse/SDK-355
## Description
Adds documentation for adding custom targeting to iOS and Fenix without blocking on the SDK.

- [x]  iOS docs
- [x] Fenix docs

Will keep this as a draft till the example PRs are merged ~~and I add the fenix docs~~:
   - ref iOS: https://github.com/mozilla-mobile/firefox-ios/pull/8952
   - ref fenix: https://github.com/mozilla-mobile/fenix/pull/20642

### CODE REVIEW

File: docs/ios-custom-targeting.md
Comment (travis79) [inline docs/ios-custom-targeting.md:10]:
```suggestion
The Nimbus SDK exposes a new `customTargetingAttributes` parameter in its initializer that is a `[String:String]` map. We can take advantage of this parameter to pass in new targeting attributes without modifying the Nimbus SDK at all.
```

File: docs/ios-custom-targeting.md
Comment (travis79) [inline docs/ios-custom-targeting.md:12]:
```suggestion
A current limitation is that both the key and the value of the targeting parameter are **strings**, therefore for any targeting parameters that require integer comparison, or any other richer `JEXL` expressions that cannot be done with strings, please reach out to the Nimbus SDK team if you need something beyond strings.
```

File: docs/ios-custom-targeting.md
Comment (travis79) [inline docs/ios-custom-targeting.md:17]:
```suggestion
In [AppDelegate+Experiments.swift](https://github.com/tarikeshaq/firefox-ios/blob/add-first-run-targeting/Client/Application/AppDelegate%2BExperiments.swift#L17) the map `customTargetingAttributes` will be used to add custom targeting. Simply add a new key-value pair to the map and it will be available for targeting. For example:
```

File: docs/ios-custom-targeting.md
Comment (travis79) [inline docs/ios-custom-targeting.md:23]: Nice job linking to the next steps!

**Review by travis79 — APPROVED** [review]
Just a few suggestions/nits, no need to block on a fresh review.

File: docs/fenix-custom-targeting.md
Comment (tarikeshaq) [inline docs/fenix-custom-targeting.md:17]: Reminder to myself to update this to point to main once the PR is merged (do the same for ios)

File: docs/fenix-custom-targeting.md
Comment (tarikeshaq) [inline docs/fenix-custom-targeting.md:33]: Reminder to myself to add this as well once that PR is created: (I have a commit: https://github.com/tarikeshaq/experimenter/commit/0a45096fd83cf7e172a9bff30780aee3bacd50bb, just holding off until the client PRS merged)

File: docs/fenix-custom-targeting.md
Comment (k88hudson) [inline docs/fenix-custom-targeting.md:5]: Nit, but can you use the word `attribute` v.s. parameter, since this is how it's referred to in a lot of existing documentation and in [desktop](https://firefox-source-docs.mozilla.org/browser/components/newtab/content-src/asrouter/docs/targeting-attributes.html)?

**Comment by k88hudson** [comment]
This looks great, I'd recommend cross-linking from the custom audiences pages as well with something like "If you're an [ios developer](link) or [fenix developer](link) for for how to add new client-side targeting attributes", and maybe that new targeting parameters have to ride the trains.

I think it would additionally be helpful to have some documentation for each targeting attribute (see example: https://firefox-source-docs.mozilla.org/browser/components/newtab/content-src/asrouter/docs/targeting-attributes.html#profileagecreated) that lives with the implementation, which you could link to from here. I'll file something about this

**Pull request merged**
